src/main/java/Storage.java

```diff
@@ -0,0 +1,121 @@
+    }
+
+    /**
+     * Method to begin the file reading process from history.
+     * @return
+     * @throws IOException
+     */
+    public ArrayList<Task> load() throws IOException {
+        File data = new File(filePath);
+
+        // Check if duke.txt exists, if not create file and retrieve data
+        if (!data.exists()) {
+            data.createNewFile();
+        } else {
+            Scanner read = new Scanner(data);
+            while (read.hasNext()) {
+                String currLine = read.nextLine();
+                Scanner cL = new Scanner(currLine);
+                String type = cL.next();
+                cL.next();
+                String isDone = cL.next();
+                cL.next();
+                if (type.equals("T")) {
+                    Todo toAdd = new Todo(cL.nextLine().substring(1));
+                    if (isDone.equals("1")) {
+                        toAdd.markAsDone();
+                    }
+                    arr.add(toAdd);
+                } else if (type.equals("D")) {
+                    String desc = cL.next();
+                    String add = cL.next();
+                    while (!add.equals("|")) {
+                        desc = desc + " " + add;
+                        add = cL.next();
+                    }
+                    String dtString = (cL.nextLine()).substring(1);
+                    Scanner dT = new Scanner(dtString);
+                    LocalDate date = LocalDate.parse(dT.next());
+                    Deadline curr;
+                    if (dT.hasNext()) {
+                        String duration = dT.next();
+                        curr = new Deadline(desc, date, duration);
+                    } else {
+                        curr = new Deadline(desc, date);
+                    }
+                    arr.add(curr);
+                } else if (type.equals("E")) {
+                    String desc = cL.next();
+                    String add = cL.next();
+                    while (!add.equals("|")) {
+                        desc = desc + " " + add;
+                        add = cL.next();
+                    }
+                    String dtString = (cL.nextLine()).substring(1);
+                    Scanner dT = new Scanner(dtString);
+                    LocalDate date = LocalDate.parse(dT.next());
+                    Event curr;
+                    if (dT.hasNext()) {
+                        String duration = dT.next();
+                        curr = new Event(desc, date, duration);
+                    } else {
+                        curr = new Event(desc, date);
+                    }
+                    arr.add(curr);
+                }
+            }
+        }
+        return arr;
+    }
+
+    /**
+     * Function to write data to txt file. Processes the instances and then converts to String to be written.
+     * @param array
+     * @throws IOException
+     */
+    public void saveData(ArrayList<Task> array) throws IOException {
```

Comment:
I think it is a good idea to park your toDatabaseStringRepresentation method under each of the Task classes (and with Task being an abstract class that the other subtasks can inherit from) to prevent the check for instanceof's. (Let Java polymorphism do the work for you).